docs/architecture/adr/ADR030-aws-service-operator.md

```diff
@@ -0,0 +1,29 @@
+# ADR030: AWS Service Operator
+
+## Status
+
+Pending
+
+## Context
+
+[Amazon announced](https://aws.amazon.com/blogs/opensource/aws-service-operator-kubernetes-available/) an AWS Service Operator for Kubernetes in October 2018.
+There is a need for one or more users to provision queues (potentially SQS ones) for their application to interact with.
+The AWS Service Operator consists of a container that sits in the Kubernetes cluster and monitors for custom resource types which it maps to the appropriate CloudFormation resources and deploys.
+It supports the following resources:
+* CloudFormation Template
+* DynamoDB
+* ECR Repository
+* ElastiCache
+* S3 Bucket
+* SNS Subscription
+* SNS Topic
+* SQS Queue
```

Comment:
I think it would be good having a section on alternatives considered:

```
## Alternative ways of provisioning SQS

We considered some alternative ways of provisioning SQS queues.  In particular, there is also an [AWS service broker](https://github.com/awslabs/aws-servicebroker) which provides AWS services via an [open service broker](https://www.openservicebrokerapi.org/) API.  The use of service brokers is also interesting with a view towards future work.  In particular, GOV.UK PaaS provides services via service brokers, and so there would be opportunity for reuse of components such as [paas-rds-broker](https://github.com/alphagov/paas-rds-broker).

However, the current way that you integrate the service broker API into kubernetes is via the [service catalog](https://github.com/kubernetes-sigs/service-catalog).  This is a somewhat leaky abstraction in that it adds a whole bunch of Custom Resource Definitions (CRDs) that users have to understand; it also introduces a separate command-line interface for browsing the service catalog, which increases the number of tools a user has to learn.

We also considered that paas-rds-broker is currently deployed using bosh, which is a fairly invasive tool that we don't want to adopt if we can avoid it.  However, after speaking with the GOV.UK PaaS team, they think that it would be relatively easy to dockerise paas-rds-broker and deploy it to kubernetes; so this is not a concern for our decision.
```

This is based on my understanding of previous discussions from a conversation with @chrisfarms.

Reply:
Well this is based on a bunch of Custom Resource Definitions too...

Reply:
Yes - that the only thing we've tested.  I think it's premature to say we think AWS service operator is suitable for these other things.

Reply:
(spoke to Phil, this set of CRDs is better than that set of CRDs, let's write down why tomorrow and get this bit done)